### ISSUE TITLE
getBlobToStream method hangs when MD5 validation on blob fails

### ISSUE THREAD

**Issue by gmantri** [Azure/azure-storage-node#643]
> For latest features support, please switch to [Azure Storage JavaScript SDK V10](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage). 

### Which service(blob, file, queue, table) does this issue concern?
Blob

### Which version of the SDK was used?
2.10.3

### What's the Node.js/Browser version?
10.16.3

### Steps to reproduce the issue?

- Upload a blob.

- Manually change it's content-md5 property so that it doesn't match with the md5 of the actual content.

- Download the blob with `disableContentMD5Validation` option as `false` (i.e. default).

- Download will hang.

### Have you found a mitigation/solution?
Yes

Here's the code that's causing the problem:

```
this.getBlobToStream(container, blob, writeStream, options, function (error, responseBlob, response) {
	if (error) {
		var onErrorCallback = function() {
			// If the download failed from the beginning, remove the file.
			if (fs.existsSync(localFileName) && writeStream.bytesWritten === 0) {
				fs.unlinkSync(localFileName);
			}
			callback(error, responseBlob, response);
		};
		if (!writeStream.closed) {
			writeStream.end(onErrorCallback);
		} else {
			onErrorCallback();
		}
	} else {
		callback(error, responseBlob, response);
	}
});
```
In my testing what I noticed was that `writeStream` was `opened` but it was not `writable` thus any attempt to call `end` on that does not do anything.

Considering `end` method writes to the stream, a simpler check would be to see if the stream is `writable` or not. If it is writable, then we should call `end` method on that stream.

Basically do something like the following:

```
	if (writeStream.writable) {
		writeStream.end(onErrorCallback);
	} else {
		onErrorCallback();
	}

```

**Comment by ljian3377** [comment]
First of all, this repo is now in maintence mode and we will only fix security related bugs. Please switch to the [latest storage SDK](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage) if possible.

And I am confused with your description and not sure how to reproduce it.
What's the cause, MD5 mismatch or the provided writeStream being not writable? And how do you get the writeStream?
How did you supply the mismatching MD5?
How did you create the writeStream?
Where did it hang, in your callback code or in the storage code?

Better if you can provide your code snippet.

I tried with this but didn't figure out your scenario.
```
process.env["AZURE_STORAGE_CONNECTION_STRING"] = "";

var azure = require('azure-storage');
var blobService = azure.createBlobService();
var fs = require('fs');

blobService.createContainerIfNotExists('mycontainer', {
    publicAccessLevel: 'blob'
  }, function(error, result, response) {
    if (!error) {
        // options = {
        //     contentSettings: {
        //         contentMD5: "jWESFRUmi+1AU3koHcwLqA=="
        //     }
        // };
        blobService.createBlockBlobFromLocalFile('mycontainer', 'taskblob', 'README.md', /*options,*/ function(error, result, response) {
            if (!error) {
                console.log("file uploaded");
                blobService.getBlobToStream('mycontainer', 'taskblob', fs.createWriteStream('output.txt'), function(error, result, response) {
                    if (!error) {
                        console.log("blob retrieved");
                    } else {
                        console.log(error);
                    }
                });
            } else {
                console.log(error);
            }
        });
    } else {
        console.log(error);
    }
  });

```
